modules/spring-config/src/test/java/org/mule/runtime/config/spring/InjectParamsServiceProxyTestCase.java

```diff
@@ -0,0 +1,288 @@
+
+import java.lang.reflect.Method;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class InjectParamsServiceProxyTestCase extends AbstractMuleContextTestCase {
+
+  @Rule
+  public ExpectedException expected = ExpectedException.none();
+
+  private Object augmentedParam;
+
+  @Test
+  public void notAugmentedInvocation() throws Exception {
+    BaseService service = new BasicService();
+
+    final BaseService serviceProxy = (BaseService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented();
+
+    assertThat(augmentedParam, is(true));
+  }
+
+  @Test
+  public void augmentedInvocation() throws Exception {
+    BaseService service = new AugmentedMethodService();
+
+    final BaseService serviceProxy = (BaseService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented();
+
+    assertThat(augmentedParam, sameInstance(muleContext));
+  }
+
+  @Test
+  public void namedAugmentedInvocation() throws Exception {
+    BaseService service = new NamedAugmentedMethodService();
+
+    final BaseService serviceProxy = (BaseService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented();
+
+    assertThat(augmentedParam, sameInstance(muleContext));
+  }
+
+  @Test
+  public void hiddenAugmentedInvocation() throws Exception {
+    BaseService service = new HiddenAugmentedMethodService();
+
+    final BaseService serviceProxy = (BaseService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented();
+
+    assertThat(augmentedParam, is(true));
+  }
+
+  @Test
+  public void overloadedAugmentedInvocation() throws Exception {
+    BaseOverloadedService service = new OverloadedAugmentedMethodService();
+
+    final BaseOverloadedService serviceProxy =
+        (BaseOverloadedService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented();
+
+    assertThat(augmentedParam, is(true));
+  }
+
+  @Test
+  public void overloadedAugmentedInvocation2() throws Exception {
+    BaseOverloadedService service = new OverloadedAugmentedMethodService();
+
+    final BaseOverloadedService serviceProxy =
+        (BaseOverloadedService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    serviceProxy.augmented(1);
+
+    assertThat(augmentedParam, sameInstance(muleContext));
+  }
+
+  @Test
+  public void ambiguousAugmentedInvocation() throws Exception {
+    BaseService service = new AmbiguousAugmetedMethodService();
+
+    final BaseService serviceProxy = (BaseService) createInjectProviderParamsServiceProxy(service, muleContext);
+
+    expected.expect(IllegalDependencyInjectionException.class);
+    expected
+        .expectMessage("More than one invocation candidate for for method 'augmented' in service 'AmbiguousAugmentedMethodService'");
```

Comment:
Create a static method in the production code to create the error message instead of hardcoding it here